service/controller/resource/certificates/create.go

```diff
@@ -0,0 +1,39 @@
+package certificates
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
```

Comment:
Can we add a comment somewhere why all that stuff is done? It doesn't look obvious. From the issue:

> The Kubernetes Secrets we currently use for prometheus (e.g: `$CLUSTER_ID-prometheus`) are held in the `default` namespace.
> 
> We want to run the tenant cluster Prometheus servers in a per-tenant cluster namespace (`$CLUSTER_ID-prometheus`). `prometheus-operator` does not support referencing a secret in a different namespace.
> 
> So, we need to copy the secret `default/$CLUSTER_ID-prometheus` to the tenant cluster prometheus namespace - e.g: `$CLUSTER_ID-prometheus/$CLUSTER_ID-prometheus`

Reply:
added this here: https://github.com/giantswarm/prometheus-meta-operator/pull/16